src/map/skill.c

```diff
@@ -3337,8 +3338,9 @@ static int skill_attack(int attack_type, struct block_list *src, struct block_li
 			case SR_KNUCKLEARROW:
 				if( skill->blown(dsrc,bl,dmg.blewcount,dir,0) && !(flag&4) ) {
 					short dir_x, dir_y;
-					dir_x = dirx[(dir+4)%8];
-					dir_y = diry[(dir+4)%8];
+					Assert_ret(dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX);
```

Comment:
Friend mentioned that if this asserting is triggered and returns, the mapfreeblock_lock earlier won't be unlocked, making things go out of sync.

Reply:
for this cases you can make an if with the assert condition to free up/unbock any necessarily variables/objects and then inside the if add the assert

Reply:
resolved in 8e641699a44dbc471bce34d1b90f18f5cb4ebfe7